### PR TITLE
Bug: afterSave for new models flushes unsaved data

### PR DESCRIPTION
**Pull request by shirase**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #15875

### CODE REVIEW

**Comment by samdark** [comment]
Merged. Thank you!

File: framework/db/BaseActiveRecord.php [hunk 1]
```diff
@@ -458,6 +458,12 @@ protected function createRelationQuery($class, $link, $multiple)
      */
     public function populateRelation($name, $records)
     {
+        foreach ($this->_relationsDependencies as $attribute => $relationNames) {
+            if (isset($relationNames[$name])) {
+                unset($this->_relationsDependencies[$attribute][$name]);
+            }
+        }
```
Comment (cebe) [inline framework/db/BaseActiveRecord.php]: doesn't this revert the whole idea of what `_relationsDependencies` is doing?

Comment (cebe) [reply framework/db/BaseActiveRecord.php]: the tests in https://github.com/yiisoft/yii2/pull/13618/files#diff-5c2b59d39a2ea1dc5443fb98d1325d11 do not cover eager loading. Pretty sure this gets broken by this change.

Comment (shirase) [reply framework/db/BaseActiveRecord.php]: Eager loading can have bug. Needs to be tested.

Comment (shirase) [reply framework/db/BaseActiveRecord.php]: relationsDependencies logic, do not works with eager loading initially, my pr do not change that behavior.

**Pull request merged**